server/test/controllers/HomeControllerAuthenticatedTest.java

```diff
@@ -53,4 +62,56 @@ public void testAuthenticatedSecurePage() {
     Result result = route(app, request);
     assertThat(result.status()).isEqualTo(HttpConstants.OK);
   }
+
+  @Test
+  public void testLanguageSelectorShown() {
+    Applicant applicant = createApplicantWithMockedProfile();
+    HomeController controller = instanceOf(HomeController.class);
+    Result result =
+        controller.index(addCSRFToken(fakeRequest()).build()).toCompletableFuture().join();
+    ;
+    assertThat(result.redirectLocation())
+        .contains(
+            controllers.applicant.routes.ApplicantInformationController.edit(applicant.id).url());
+  }
+
+  @Test
+  public void testLanguageSelectorNotShownOneLanguage() {
+    Applicant applicant = createApplicantWithMockedProfile();
+    Langs mockLangs = Mockito.mock(Langs.class);
+    when(mockLangs.availables()).thenReturn(ImmutableList.of(Lang.forCode("en-US")));
+
+    HomeController controller =
+        new HomeController(
+            instanceOf(LoginForm.class),
+            instanceOf(ProfileUtils.class),
+            instanceOf(MessagesApi.class),
+            instanceOf(HttpExecutionContext.class),
+            mockLangs);
+    Result result =
+        controller.index(addCSRFToken(fakeRequest()).build()).toCompletableFuture().join();
+    assertThat(result.redirectLocation())
+        .contains(
+            controllers.applicant.routes.ApplicantProgramsController.index(applicant.id).url());
+  }
+
+  @Test
```

Comment:
I'd also test where the locale is set to something other than the single-configured language (edge case, but good to specify the behavior).

Reply:
done.